### PR TITLE
Refactor repository to prepare for more packages

### PR DESCRIPTION
**Pull request by lluiscampos**
Namely:
* Move the recipes into a subfolder with the mender-client prefix
* Move the source checkout from the Dockerfile to the runtime script
* Pass arguments to the entrypoint script so that it can be reused

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -29,30 +29,9 @@ RUN wget -q https://dl.google.com/go/go$GOLANG_VERSION.linux-amd64.tar.gz \
...
-      # It is called "mender" for 2.2.x and earlier
-      cp support/mender-client.service ${debiandir}/mender-client.service ||  \
-      cp support/mender.service ${debiandir}/mender-client.service;           \
-    done
+COPY recipes /recipes
 
 # Prepare the deb-package script
 COPY mender-deb-package /usr/local/bin/
-ENTRYPOINT bash /usr/local/bin/mender-deb-package
+ENTRYPOINT  ["bash", "/usr/local/bin/mender-deb-package"]
```
Comment (oleorhagen) [inline Dockerfile:37]:
```suggestion
ENTRYPOINT  /usr/local/bin/mender-deb-package
```
How about we just give the script a hashbang?

File: mender-deb-package [hunk 2]
```diff
@@ -85,7 +116,13 @@ copy_packages_to_output() {
 
 verify_output_directory_exists
 
-extract_client_version
+verify_script_arguments $@
```
Comment (oleorhagen) [inline mender-deb-package:119]:
```suggestion
verify_script_arguments "$@"
```

Comment (oleorhagen) [inline mender-deb-package:119]:
```suggestion
verify_script_arguments "$@"
```

**Review by oleorhagen — APPROVED** [review]
So much Yes!

File: mender-deb-package [hunk 3]
```diff
@@ -21,25 +21,56 @@ verify_output_directory_exists() {
...
+  VERSION=$3
+}
+
+checkout_repo() {
+  go get -d $REPO_URL
+  cd /root/go/src/$REPO_URL
+  git checkout $VERSION
+}
+
+get_package_version() {
```
Comment (kacf) [inline mender-deb-package:39]: Would consider using `get_deb_version` to match the variable name it's storing it in.

File: mender-deb-package (hunk 2 above)
Comment (kacf) [reply mender-deb-package:119]: +1

Comment (kacf) [reply mender-deb-package:119]: +1

File: Dockerfile (hunk 1 above)
Comment (kacf) [reply Dockerfile:37]: Yes, it even has one, but it has a typo, missing the `!`

**Review by kacf — APPROVED** [review]
Looks good, just minor adjustments.

File: Dockerfile (hunk 1 above)
Comment (lluiscampos) [reply Dockerfile:37]: I'll fix the typo, thanks.

With regards to the syntax change. We need to use the exec form (i.e. `[...]`) for then `docker run` be able to pass parameters to the script. See [Docker documentation](https://docs.docker.com/engine/reference/builder/#entrypoint).

**Pull request merged**